src/api/environment.cc

```diff
@@ -399,6 +399,33 @@ Local<Context> NewContext(Isolate* isolate,
   return context;
 }
 
+Local<Context> NewContext(Environment* env,
+                          Local<ObjectTemplate> object_template,
+                          bool initialize) {
+  v8::EscapableHandleScope scope(env->isolate());
+  auto context = Context::New(env->isolate(), nullptr, object_template);
+  if (context.IsEmpty()) return context;
+
+  if (initialize && !InitializeContext(context)) {
+    return Local<Context>();
+  }
```

Comment:
Up until here, this mostly duplicates the existing `NewContext()` above. Can you call that function instead of copying the lines?

Reply:
I am not sure how. The feature is to allow embedders to have something that allow ``v8::Context::New`` to be inspectable. The global object created will have only the V8 builtins (aside what is defined in the ``ObjectTemplate``). In contrast ``node::InitializeContext`` would add and remove properties of the global object related to node. For that reason, the additional argument ``initialize`` allows assigning the new context to environment without invoking ``node::InitializeContext``.